dispatcher.go

```diff
@@ -80,6 +81,7 @@ func NewDispatcher(cfg Config) *Dispatcher {
 
 	meter, stopMeter := cfg.Metrics.scope(cfg.Name, logger)
 	cfg = addObservingMiddleware(cfg, meter, logger, extractor)
+	cfg = addFirstOutboundMiddleware(cfg)
```

Comment:
should this be added before the ObservingMiddleware?

Reply:
Indeed. By being added last, it is the outermost middleware and will execute first.